libraries/rush-lib/src/logic/buildCache/AdditionalFilesHasher.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import { Import } from '@rushstack/node-core-library';
+import crypto from 'crypto';
+import fs from 'fs';
+import path from 'path';
+
+import type { IOptions } from 'glob';
+
+const glob: typeof import('glob') = Import.lazy('glob', require);
+
+const globAsync = (pattern: string, options: IOptions = {}): Promise<string[]> => {
+  return new Promise((resolve, reject) => {
+    glob(pattern, options, (err, matches) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+      resolve(matches);
+    });
+  });
+};
+
+export class AdditionalFilesHasher {
+  public static async getFileHashedFilesMap(
+    globPatterns: string[],
+    cwd: string
+  ): Promise<Map<string, string>> {
+    const filePaths: Set<string> = await this._expandGlobPatterns(globPatterns, cwd);
+
+    return await this._calculateFileHashes(filePaths, cwd);
+  }
+
+  private static async _expandGlobPatterns(globPatterns: string[], cwd: string): Promise<Set<string>> {
+    const allMatches: Set<string> = new Set<string>();
+    for (const pattern of globPatterns) {
+      const matches: string[] = await globAsync(pattern, {
+        cwd,
+        nodir: true,
+        // We want to keep path's type unchanged,
+        // i.e. if the pattern was a  relative path, then matched paths should also be relative paths
+        //      if the pattern was an absolute path, then matched paths should also be absolute paths
+        //
+        // We are doing this because these paths are going to be used to calculate a hash for the build cache and some users
+        // might choose to depend on global files (e.g. `/etc/os-release`) and some might choose to depend on local files
+        // (e.g. `../path/to/workspace/file`)
+        //
+        // In both cases we want that path to the resource would be the same on all machines,
+        // regardless of what is the current working directory.
+        //
+        // That being said, we want to keep `realpath` and `absolute` options here as false:
+        realpath: false,
+        absolute: false
+      });
+
+      matches.forEach((match) => allMatches.add(match));
+    }
+
+    if (allMatches.size === 0) {
+      throw new Error(
+        `Couldn't find any files matching provided glob patterns: ["${globPatterns.join('", "')}"].`
+      );
```

Comment:
I personally would love to give more instructions how to fix this. E.g. include path to `rush-project.json`. 
But I'm not sure whether the pathname of `rush-project.json` will not change in the future versions of rush and whether there is a wish to add extensive suggestions in the first place.

Reply:
Regarding the check it self - I personally would love to provide more options for the maintainers, e.g. mark whether a pattern is optional or not. 

But the same could be said about `dependsOnEnvVars` setting, and I wanted to keep things aligned.

And if we'll want to change this - maybe we should for both options? 🤔